irohad/consensus/yac/impl/yac_gate_impl.cpp

```diff
@@ -48,104 +39,102 @@ namespace iroha {
             block_loader_(std::move(block_loader)),
             consensus_result_cache_(std::move(consensus_result_cache)),
             log_(logger::log("YacGate")) {
-        block_creator_->on_block().subscribe(
-            [this](auto block) { this->vote(block); });
+        block_creator_->on_block().subscribe([this](auto block) {
+          // TODO(@l4l) 24/09/18 IR-1717
+          // update BlockCreator iface according to YacGate
+          this->vote(
+              std::shared_ptr<shared_model::interface::Proposal>(
+                  (shared_model::proto::ProtoProposalFactory<
```

Comment:
Proto should not be used even as a temporary solution. Consider making a dummy interface implementation in the same file.